src/main/resources/ZoneCode/ZoneMacro.xml

```diff
@@ -297,8 +315,11 @@
       <code>{{velocity}}
 {{html clean="false"}}
 #set($docToDisplay = "$!xcontext.macro.params.document")
+#if($xwiki.hasAccessLevel('edit', $xcontext.user, $docToDisplay))
+  #set($editZone = "editable")
+#end 
 #if("$!docToDisplay"!="" &amp;&amp; $xwiki.exists("$!docToDisplay"))
-  &lt;div class="section_container"&gt;
+  &lt;div class="section_container $editZone"&gt;
```

Comment:
If the user does not have `edit` rights, the class attribute will be `section_container $editZone`, since `$editZone` will be `null`. To avoid that, make sure to use `$!editZone` instead which will be an empty string when `$editZone` is `null`.